test/manual/x509_auth.test.ts

```diff
@@ -0,0 +1,134 @@
+import { expect } from 'chai';
+import { ConnectionString } from 'mongodb-connection-string-url';
+
+import {
+  MongoClient,
+  type MongoClientOptions,
+  MongoServerError,
+  MongoServerSelectionError
+} from '../../src';
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const connectionString = new ConnectionString(process.env.MONGODB_URI!);
+
+describe('x509 Authentication', function () {
+  let client: MongoClient;
+  const validOptions: MongoClientOptions = {
+    tls: true,
+    tlsCertificateKeyFile: process.env.SSL_KEY_FILE,
+    tlsCAFile: process.env.SSL_CA_FILE,
+    authMechanism: 'MONGODB-X509' as const,
+    authSource: '$external'
+  };
+
+  this.afterEach(() => {
+    return client?.close();
+  });
+
+  context('When the user provides a valid certificate', function () {
+    before('create x509 user', createX509User);
+    after('drop x509 user', dropX509User);
+
+    it('successfully authenticates using x509', async function () {
+      client = new MongoClient(connectionString.toString(), validOptions);
+      const result = await client
+        .db('aws')
+        .collection('aws_test')
```

Comment:
not that this matters but if you're making the other changes might as well: `aws` -> `x509`